isolated script idempotency

bitscuit opened the pull request.

bluzarraga commented on csmaps_utils.sh:
```diff
@@ -0,0 +1,162 @@
+
+OC=oc
+YQ=yq
+CM_NAME="mock-cs-maps"
```
assuming this should be changed back to common-service-maps

bitscuit replied:
I am leaning towards keeping it as `mock-cs-maps` since this file is primarily for testing and storing the functions to manipulate configmaps, not actually to be used as-is. If this script is included by another script, the CM_NAME can be overwritten by the importing script anyways

bitscuit commented on csmaps_utils.sh:
```diff
@@ -0,0 +1,162 @@
+    msg "[INFO] ${1}"
+}
+
+# create_empty_csmaps
+
+# insert_control_ns
+
+# update_tenant "tenant1"
+# echo "List of namespaces in tenant1: "
+# read_tenant_from_csmaps "tenant1"
+
+# update_tenant "tenant1" "ns1 ns2"
+# echo "List of namespaces in tenant1: "
+# read_tenant_from_csmaps "tenant1"
+
+# update_tenant "tenant2" "ns3"
+# echo "List of namespaces in tenant2: "
+# read_tenant_from_csmaps "tenant2"
```
keeping this here for quick testing of the script

bluzarraga commented on csmaps_utils.sh:
```diff
@@ -0,0 +1,162 @@
+function insert_control_ns() {
+    local current_yaml=$("${OC}" get -n kube-public cm "$CM_NAME" -o yaml | "${YQ}" '.data.["common-service-maps.yaml"]')
+
+    current=$(echo "$current_yaml" | "${YQ}" .'controlNamespace')
```
does the period need to be inside the ''? '.controlNamespace' instead of .'controlNamespace'

bitscuit replied:
good catch. Weird that it still worked though

bluzarraga commented on csmaps_utils.sh:
```diff
@@ -0,0 +1,162 @@
+function read_tenant_from_csmaps() {
+    local map_to_cs_ns=$1
+    local current_yaml=$("${OC}" get -n kube-public cm "$CM_NAME" -o yaml | "${YQ}" '.data.["common-service-maps.yaml"]')
+    local tenant_ns_list=$(echo "$current_yaml" | "${YQ}" eval '.namespaceMapping[] | select(.map-to-common-service-namespace == "'${map_to_cs_ns}'").requested-from-namespace' | awk '{ print $2 }')
```
you might need to truncate newline characters here otherwise the list is passed on like:  "ns1\nns2\nns3 ...etc". You can use `| tr '\n' ' '` after the awk command

bitscuit replied:
`tenant_ns_list` does end up being newline delimited here. I don't think whether it's newline or space delimited impacts anything.

bitscuit commented:
The git diff looks a bit wonky, but in summary:
1. I copied the csmaps_utils function to isolate.sh
1. optimized gather_csmaps_ns
1. removed mapping_topology function since the csmaps_utils functions can be used instead to construct the common-service-maps
1. merged abort_check into prereq
1. removed many global variables and localized any variables declared in functions
1. optimized licensing configmap migration

Contrary to what the diff shows, I did not make changes to removeNSS nor did I remove cleanupCSOperators. If you look at what is in the scripts branch and what is in this branch, you can see there is no difference.

bluzarraga commented:
/lgtm
